samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -39,12 +43,46 @@ static int get_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 		if (request_value[1] == 0x03) {
 			/* Request for feature report */
 			if (request_value[0] == REPORT_ID_USER_CONFIG) {
-				static u8_t status;
-
-				status = atomic_get(&forward_status);
-				*len = sizeof(status);
-				*data = (u8_t *)(&status);
-
+				size_t length = *len;
+				u8_t *buffer = *data;
+
+				struct config_channel_frame frame;
+				int pos = config_channel_report_parse(buffer, length, &frame, true);
+				if (pos < 0) {
+					LOG_WRN("Could not parse report");
+					return pos;
+				}
+
+				bool is_fetch = (frame.status == FORWARD_STATUS_FETCH);
+
+				if (is_fetch) {
+					__ASSERT_NO_MSG(frame.event_data_len == 0);
+
+					if (atomic_get(&fetch_done)) {
```

Comment:
I think we should check that fetch was for the same item
host can request one thing but subsequent messages can have something else (due to a bug or attempt to crash a device)

Reply:
done